impexp-core/src/main/java/org/citydb/citygml/deleter/database/DeleteManager.java

```diff
@@ -261,7 +266,23 @@ private void doDelete(Select select, long hits) throws SQLException {
 		}
 	}
 
-	private void doTerminate(Select select) throws SQLException {
+	private void doTerminate(Select select) throws SQLException, IOException {
+		if (deleteLogger != null) {
+			try (PreparedStatement stmt = databaseAdapter.getSQLAdapter().prepareStatement(select, connection);
+				 ResultSet rs = stmt.executeQuery()) {
+				while (rs.next()) {
+					long id = rs.getLong("id");
+					int objectClassId = rs.getInt("objectclass_id");
+					String gmlId = rs.getString("gmlid");
+					AbstractObjectType<?> objectType = schemaMapping.getAbstractObjectType(objectClassId);
+
+					deleteLogger.write(objectType != null ? objectType.getPath() : String.valueOf(objectClassId), id, gmlId);
```

Comment:
Hm, i think we should not allow terminating NULL-objectType (e.g. ADE classes which are registered in DB, but not supported by impexp), in order to keep consistent behaviour in delete and export processes.

Reply:
Terminate fails now in case the `objectType` cannot be derived is thus is null (see fefd0cf32f24162d0f2a95828cf74125bb90f6db).

Reply:
Looks OK to me for this PR. But this would work only when the delete logger is enabled. I think we could do some furhter tweaks in a separate PR.

Reply:
Yes, we should definitely put this discussion into a separate PR because this PR is just about adding a delete log.

Well, the reason for having this restriction for the import and export operation is simply because we need additional code to be able to parse and write ADE datasets. When deleting, we do not need to parse/write ADE datasets. So I am not convinced that we really should restrict the delete operation in this way.

Reply:
Ok, one more note on this: I think the delete operation already behaves like you expect (even before this PR). I tested by registering the TestADE and importing a TestADE dataset. Afterwards, I removed the TestADE extension from the Importer/Exporter (so that the TestADE is still registered in DB but not supported by the Importer/Exporter).

Afterwards, I executed a delete operation. TestADE features are not deleted/terminated in this case.

Ok, will merge this PR now.